tests/unit/test_job_retry.py

```diff
@@ -240,3 +243,69 @@ def test_raises_on_job_retry_on_result_with_non_retryable_jobs(client):
         ),
     ):
         job.result(job_retry=google.api_core.retry.Retry())
+
+
+def test_query_and_wait_retries_job():
+    """
+    Specific test for retrying DDL queries with "jobRateLimitExceeded" error:
+    https://github.com/googleapis/python-bigquery/issues/1790
+    """
+    freezegun.freeze_time(auto_tick_seconds=100)
+    client = mock.create_autospec(Client)
+    client._call_api.__name__ = "_call_api"
+    client._call_api.__qualname__ = "Client._call_api"
+    client._call_api.__annotations__ = {}
+    client._call_api.__type_params__ = ()
+    client._call_api.side_effect = (
+        google.api_core.exceptions.BadRequest("jobRateLimitExceeded"),
```

Comment:
I don't think our retry code looks at the "message", right, only errors. Let's make sure we populate those.

https://github.com/googleapis/python-api-core/blob/17ff5f1d83a9a6f50a0226fb0e794634bd584f17/google/api_core/exceptions.py#L135

Reply:
I would also recommend having first request "succeed" but has `"jobComplete": False,`. That way we can be more sure that we're testing the job retry mechanism and not the api call level retries.

Reply:
```
```suggestion
        {
            "jobReference": {
                "projectId": "response-project",
                "jobId": "abc",
                "location": "response-location",
            },
            "jobComplete": False,
        },
```